entity-framework/core/modeling/value-conversions.md

```diff
@@ -2,145 +2,746 @@
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder
+                    .Entity<Rider>()
+                    .Property(e => e.Mount)
+                    .HasConversion(
+                        v => v.ToString(),
+                        v => (EquineBeast)Enum.Parse(typeof(EquineBeast), v));
+            }
+-->
+[!code-csharp[ExplicitConversion](../../../samples/core/Modeling/ValueConversions/EnumToStringConversions.cs?name=ExplicitConversion)]
 
-```csharp
-public class Rider
-{
-    public int Id { get; set; }
-    public EquineBeast Mount { get; set; }
-}
-
-public enum EquineBeast
-{
-    Donkey,
-    Mule,
-    Horse,
-    Unicorn
-}
-```
+> [!NOTE]
+> A `null` value will never be passed to a value converter. A null in a database column is always a null in the entity instance, and vice-versa. This makes the implementation of conversions easier and allows them to be shared amongst nullable and non-nullable properties. See [GitHub issue #13850](https://github.com/dotnet/efcore/issues/13850) for more information.
 
-Then conversions can be defined in `OnModelCreating` to store the enum values as strings (for example, "Donkey", "Mule", ...) in the database:
+## Pre-defined conversions
 
-```csharp
-protected override void OnModelCreating(ModelBuilder modelBuilder)
-{
-    modelBuilder
-        .Entity<Rider>()
-        .Property(e => e.Mount)
-        .HasConversion(
-            v => v.ToString(),
-            v => (EquineBeast)Enum.Parse(typeof(EquineBeast), v));
-}
-```
+EF Core contains many pre-defined conversions that avoid the need to write conversion functions manually. Instead, EF Core will pick the conversion to used based on the property type in the model and the requested database provider type.
 
-> [!NOTE]
-> A `null` value will never be passed to a value converter. This makes the implementation of conversions easier and allows them to be shared amongst nullable and non-nullable properties.
+For example, enum to string conversions are used as an example above, but EF Core will actually do this automatically when the provider type is configured as `string` using the generic type of <xref:Microsoft.EntityFrameworkCore.Metadata.Builders.PropertyBuilder.HasConversion%2A>:
+
+<!--
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                modelBuilder
+                    .Entity<Rider>()
+                    .Property(e => e.Mount)
+                    .HasConversion<string>();
+            }
+-->
+[!code-csharp[ConversionByClrType](../../../samples/core/Modeling/ValueConversions/EnumToStringConversions.cs?name=ConversionByClrType)]
+
+The same thing can be achieved by explicitly specifying the database column type. For example, if the entity type is defined like so:
```

Comment:
Provide both the Fluent API and the Data Annotations alternatives via tabs, like elsewhere?